### ISSUE TITLE
Enabling NewCops in rubocop.yml does not appear to add files to exclusion with --auto-gen-only-exclude

### ISSUE THREAD

**Issue by henrytseng** [rubocop-hq/rubocop#8776]
When generating `rubocop_todo.yml` it looks enabling both new parameter `NewCops` and `--auto-gen-only-exclude` does not add files violating new rules to the exclude list.  

When adding linting to an existing project with a large code base existing files can be added to `rubocop_todo.yml`.  This makes adopting code linting easier.  With `--auto-gen-only-exclude` on and a large `--exclude-limit` new cops should be added to the `Excluded` lists under each cop in `rubocop.yml`.  

--------

## Expected behavior

Expect RuboCop to add to the `Excluded` lists in `rubocop.yml` when `NewCops` is enabled.  

## Actual behavior

Currently `NewCops` flags offenses but does not add to the `Excluded` list.  

## Steps to reproduce the problem

1. Add files with offenses included in NewCops
2. Run `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 1000`
3. Then with newly generated `rubocop_todo.yml` run `rubocop .`

## RuboCop version

```
# rubocop -V
0.91.1 (using Parser 2.7.1.4, rubocop-ast 0.4.2, running on ruby 2.6.3 x86_64-linux-musl)
```

**Comment by dvandersluis** [comment]
Do you have more than 1000 violations of the cop? If you surpass the exclude limit, the cop will be explicitly disabled in `.rubocop_todo.yml`.

I tried to replicate this, with and without surpassing the exclude limit.

Under exclude limit:

```yaml
# Offense count: 2
# Cop supports --auto-correct.
Style/StringConcatenation:
  Exclude:
    - 'test.rb'
    - 'test2.rb'
```

Over exclude limit:

```yaml
# Offense count: 2
# Cop supports --auto-correct.
Style/StringConcatenation:
  Enabled: false
```

What are you seeing happen?

**Comment by henrytseng** [comment]
I didn't have more than 1000 violations on that cop.  I can keep my eyes on the issue but if you're not able to reproduce perhaps its something else that I am doing.

**Comment by dvandersluis** [comment]
What ends up in `.rubocop_todo.yml` for the cop you're talking about?

**Comment by henrytseng** [comment]
Made a number of changes to the code base and adopted the Relaxed.Ruby.Style cops.  I can't seem to bisect the commit that generated the same errors.  

I'm seeing `Layout/SpaceBeforeBlockBraces` does not appear at all in `rubocop_todo.yml` but I'm seeing 134 pop up directly running `rubocop`.

**Comment by henrytseng** [comment]
Okay, just just saw it happen again.

1. Ran `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 2000`
2. Seeing `rubocop_todo.yml`
```
# This configuration was generated by
# `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 2000`
# on 2020-10-06 16:56:51 UTC using RuboCop version 0.91.1.

...
# Offense count: 274
# Cop supports --auto-correct.
# Configuration parameters: EnforcedStyleForEmptyBraces.
# SupportedStyles: space, no_space, compact
# SupportedStylesForEmptyBraces: space, no_space
Layout/SpaceInsideHashLiteralBraces:
  EnforcedStyle: space
```

And it looks like my original `rubocop.yml` does not contain that "Layout/SpaceInsideHashLiteralBraces".  Running `rubocop` produces 410 offenses.  

Am I doing something wrong?

**Comment by henrytseng** [comment]
Could I make a feature request on this issue?  I think this issue could be mitigated if there were some way to use the CLI to append to the existing `rubocop_todo.yml` excludes with any new issues.

**Comment by marutosi** [comment]
I think it is Layout/SpaceInsideHashLiteralBraces bug #8938.

**Comment by jshah** [comment]
I am seeing this as well and not only for `Layout/SpaceInsideHashLiteralBrace`.

When I run `bundle exec rubocop`, I see violations so then I run `bundle exec rubocop --auto-gen-config --exclude-limit 999999` to add them to my `.rubocop_todo.yml`. This does not ignore all violations and my `bundle exec rubocop` continues to fail.

**Comment by bbatsov** [comment]
Can't reproduce this on 1.89 - the todo file gets proper `Exclude` lists in both scenarios described here. The underlying bug was in the style detection of some `ConfigurableEnforcedStyle` cops, fixed by #9280 back in 1.7.0, so I'm closing this. The idea of appending new offenses to an existing todo file instead of regenerating it is a separate ask - if anyone still wants that, please open a dedicated issue.
